packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -312,14 +315,64 @@ export class PHPRequestHandler<PHP extends BasePHP> {
 			),
 			this.rewriteRules
 		);
-		const fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
-		if (!seemsLikeAPHPRequestHandlerPath(fsPath)) {
-			return this.#serveStaticFile(
-				await this.processManager.getPrimaryPhp(),
-				fsPath
-			);
+
+		const primaryPhp = await this.getPrimaryPhp();
+
+		let fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
+
+		// We can only satisfy requests for directories with a default file
+		// so let's first resolve to a default path when available.
+		if (primaryPhp.isDir(fsPath)) {
+			if (!fsPath.endsWith('/')) {
+				return new PHPResponse(
+					301,
+					{ Location: [`${requestedUrl.pathname}/`] },
+					new Uint8Array(0)
+				);
+			}
+			const localDefaultPath = joinPaths(fsPath, 'index.php');
+			if (primaryPhp.isFile(localDefaultPath)) {
+				fsPath = localDefaultPath;
+			}
```

Comment:
Should we check for `index.html` if `index.php` doesn't exist?

Reply:
Added in [f655598](https://github.com/WordPress/wordpress-playground/pull/1417/commits/f655598dc6d469bf02e78f1ff34da9a5cce054d9)